internal/node/process_mgr.go

```diff
@@ -9,34 +11,45 @@ import (
 // (e.g. firecracker) is employed. Note that agent processes are created asynchronously -before- any
 // workloads are deployed to them, so a workload manager can never explicitly tell a process manager
 // to create an individual process
-type ProcessManager interface {
-	// Starts the process manager. This allows the process manager to create a pool of agents however it sees fit
-	Start(ProcessSubscriber) error
-	// Stops the process manager and all running processes within
-	Stop() error
-	// Tells the process manager to associate the deployment request with the given workload ID, and perform any
+type ProcessManager interface { // FIXME-- rename to AgentManager?
+	// Returns a list of agent processes in an implementation-agnostic format
+	// FIXME-- rename to ListRunning()?
+	ListProcesses() ([]AgentInfo, error)
+
+	// Returns a list of pooled agent processes in an implementation-agnostic format
+	// FIXME-- rename to ListWarm()?
+	ListPool() ([]AgentInfo, error)
+
+	// Lookup a deploy request by agent id. Returns nil when attempting to lookup an "unprepared" workload
+	Lookup(agentID string) (*agentapi.DeployRequest, error)
+
+	// Associate a deploy request with the given workload id, and perform any
 	// just in time initialization of resources if necessary
-	PrepareWorkload(string, *agentapi.DeployRequest) error
-	// Requests that the process manager terminate the process with the given ID
-	StopProcess(string) error
-	// Performs a lookup on a process to return its associated deployment request. This will return nil if done
-	// on an "unprepared" workload
-	Lookup(string) (*agentapi.DeployRequest, error)
-	// Returns a list of processes in an implementation-agnostic format
-	ListProcesses() ([]ProcessInfo, error)
+	PrepareWorkload(agentID string, request *agentapi.DeployRequest) error
+
+	// Start the process manager and allocate a pool of agents based on an implementation-specific
+	// strategy, delegating callbacks to the given process subscriber
+	Start(delegate AgentDelegate) error
+
+	// Stop the process manager and gracefully shutdown all agents in the pool
+	Stop() error
+
+	// Terminate a running agent process with the given ID
+	StopProcess(agentID string) error
 }
 
-// A process subscriber is any struct that wishes too be notified when a process manager has
-// finished starting a process
-type ProcessSubscriber interface {
-	// Indicates that the process is now ready, and can be "prepared" for deployment
-	OnProcessReady(workloadId string)
+// An agent delegate is any struct that wishes to be notified when the configured agent process
+// manager has successfully started an agent
+type AgentDelegate interface {
```

Comment:
please put this name back. AgentDelegate doesn't actually describe what this is. It is *not* an agent delegate, it's a process manager delegate.